spinnaker-monitoring-daemon/tests/spectator_metric_transformer_test.py

```diff
@@ -0,0 +1,529 @@
+                 ])},
+            ])}
+        })
+
+  def test_change_tag_to_type_int(self):
+    self.do_test(
+        textwrap.dedent("""\
+            controller.invocations:
+              tags:
+                - controller
+                - method
+                - statistic
+                - status
+
+              transform_tags:
+                - from: statusCode
+                  to: statusCode
+                  type: INT
+        """),
+
+        {'controller.invocations': {
+            'kind': 'Timer',
+            'values': [
+                {'values': [{'t': 1540318956420, 'v': 300130409.0}],
+                 'tags': [
+                     {'key': 'controller', 'value': 'ClusterController'},
+                     {'key': 'method', 'value': 'getServerGroup'},
+                     {'key': 'statistic', 'value': 'totalTime'},
+                     {'key': 'status', 'value': '2xx'},
+                     {'key': 'statusCode', 'value': '200'},
+                 ]}
+            ]},
+        },
+
+        {'controller.invocations': {
+            'kind': 'Timer',
+            'values': [
+                {'values': [{'t': 1540318956420, 'v': 300130409.0}],
+                 'tags': [
+                     {'key': 'controller', 'value': 'ClusterController'},
+                     {'key': 'method', 'value': 'getServerGroup'},
+                     {'key': 'statistic', 'value': 'totalTime'},
+                     {'key': 'status', 'value': '2xx'},
+                     {'key': 'statusCode', 'value': 200},
+                 ]}
+            ]},
+        },
+    )
+
+  def test_decompose_tag(self):
+    self.do_test(
+        textwrap.dedent("""\
+            executionCount:
+              tags:
+                - status
+
+              transform_tags:
+                - from: agent
+                  to: [provider, account, region, agent]
+                  type: [STRING, STRING, STRING, STRING]
+                  extract_regex: '([^/]+)/(?:([^/]+)/(?:([^/]+)/)?)?(.+)'
+        """),
+
+        {'executionCount': {
+            'kind': 'Counter',
+            'values': sorted([{
+                'values': [{'t': 1540318956422, 'v': 23258.0}],
+                'tags': [
+                    {'key': 'agent',
+                     'value': 'com.netflix.spinnaker.clouddriver.google.provider.GoogleInfrastructureProvider/my-google-account/australia-southeast1/GoogleSubnetCachingAgent'},
+                    {'key': 'status', 'value': 'success'}
+                ]
+            }, {
+                'values': [{'t': 1540318956422, 'v': 9.0}],
+                'tags': [
+                    {'key': 'agent',
+                     'value': 'com.netflix.spinnaker.clouddriver.appengine.provider.AppengineProvider/my-appengine-account/AppenginePlatformApplicationCachingAgent'},
+                    {'key': 'status', 'value': 'failure'},
+                ]
+            },
+                             ])},
```

Comment:
Indentation looks off here (and for a number of closing braces in this file)

Reply:
Agreed, but this is what the linter wanted and it helps keep the width a bit down so I didnt argue with it. The ])} here lines up with the sorted( statement it is closing.